database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -0,0 +1,443 @@
+                if (isTableEmpty(tableName, connection)) {
+                    String resetAutoIncrementSQL = "ALTER TABLE " + tableName + " AUTO_INCREMENT = 1";
+                    statement.executeUpdate(resetAutoIncrementSQL);
+                }
+
+                String insertSQL = generateInsertSQL(entity, tableName);
+
+                LOG.info("Executing SQL: {}", insertSQL);
+
+                int affectedRows = statement.executeUpdate(insertSQL, Statement.RETURN_GENERATED_KEYS);
+                if (affectedRows == 0) {
+                    LOG.error("Creating record failed in table: {}, {}", tableName, insertSQL);
+                    throw new SQLException("Creating record failed, no rows affected.");
+                }
+
+                try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                    if (generatedKeys.next()) {
+                        entity.setId(generatedKeys.getInt(1));
+                    } else {
+                        LOG.error("Creating record failed in table: {}, {}", tableName, insertSQL);
+                        throw new SQLException("Creating record failed, no ID obtained.");
+                    }
+                }
+            } else {
+                throw new DatabaseDoesNotExistException(TABLE_NOT_EXIST + ": " + tableName);
+            }
+        } catch (SQLException e) {
+            LOG.error(UNABLE_ADD_NEW_RECORD + ": {}, {}", tableName, e.getMessage());
+            throw new DatabaseOperationException(UNABLE_ADD_NEW_RECORD + ": " + e.getMessage());
+        }
+
+        return entity;
+    }
+
+    @Override
+    public <T extends BaseEntity> T updateRecordInTable(T entity, Integer id) {
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        String tableName = entityClass.getSimpleName();
+
+        String updateSQL = generateUpdateSQL(entity, tableName, id);
+
+        LOG.info("Executing SQL: {}", updateSQL);
+
+        try (Connection connection = DriverManager.getConnection(databaseURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement()) {
+
+            int affectedRows = statement.executeUpdate(updateSQL);
+            if (affectedRows == 0) {
+                LOG.error("Updating record failed in table {}: {}", tableName, updateSQL);
+                throw new IdDoesNotExistException(ENTITY_IS_NOT_FOUND);
+            }
+        } catch (SQLException e) {
+            LOG.error("Unable to update record in table {}: {}", tableName, e.getMessage());
+            throw new DatabaseOperationException("Unable to update record in table " + ": " + tableName + ", " +
+                    e.getMessage());
+        }
+
+        return entity;
+    }
+
+    @Override
+    public boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, Integer id) {
+        String tableName = entityClass.getSimpleName();
+        String deleteRecordSQL = "DELETE FROM " + tableName + " WHERE id = " + id;
```

Comment:
Please use `Limit` value from `Settings`. See how it works in `JsonDatabaseServicegetAllRecordsFromTable`.